pkg/gadgets/socket-collector/types/socket-collector.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+type Event struct {
```

Comment:
Is this really an event? I ask because the socket-collector is not event-based, so it could cause some confusion.

Reply:
Good point. I was just trying to keep gadgets uniform but as you said, socket-collector is not even event-based as dns. 

I thought to change the name to `Socket` as it was before but it will result in a `Socket` data structure that contains an `eventtypes.Event` one which mmm does not convince me completely. In addition, `eventtypes.Event` fields like `Err` does not make sense for this kind of gadgets that are not streamed.

What if we split up the `eventtypes.Event`? Maybe into one named `Kubernetes` with `Node`, `Namespace`, `Pod` and `Container` fields. Then, I would create another one named `Event` which includes `Kubernetes` and adds `Err` and (maybe) `Notice` fields. This new data structure would replace the old `eventtypes.Event`. BTW, I am not sure where to put `Ready` and `Host` to be honest 😅 .

Starting from that, the DNS gadget can create its own (Maybe `DnsEvent`) based on `Event` as it was before, while socket-collector would create one based on `Kubernetes` and adds its own fields, resulting in something that would finally deserve to be called `Socket`. 

Does it make sense for you? I hope it was not too confusing.

cc\ @alban

Reply:
We could do that, but at the cost of more complexity. I think I'm ok either way.

>What if we split up the eventtypes.Event? Maybe into one named Kubernetes with Node, Namespace, Pod and Container fields. Then, I would create another one named Event which includes Kubernetes and adds Err and (maybe) Notice fields. This new data structure would replace the old eventtypes.Event.

It sounds logical.

> BTW, I am not sure where to put Ready and Host to be honest sweat_smile .

`Ready` only makes sense for streams and `Host` would belong in your `Kubernetes` struct along with Node, Namespace, Pod and Container.

Reply:
@mauriciovasquezbernal what do you think?

Reply:
Issue https://github.com/kinvolk/inspektor-gadget/issues/250 was created to track this task.